cloud/scope/cluster.go

```diff
@@ -100,56 +102,79 @@ func (s *ClusterScope) Authorizer() autorest.Authorizer {
 	return s.AzureClients.Authorizer
 }
 
-// Network returns the cluster network object.
-func (s *ClusterScope) Network() *infrav1.Network {
-	return &s.AzureCluster.Status.Network
-}
-
 // PublicIPSpecs returns the public IP specs.
 func (s *ClusterScope) PublicIPSpecs() []azure.PublicIPSpec {
+	var controlPlaneOutboundIP azure.PublicIPSpec
+	if s.IsAPIServerPrivate() {
+		controlPlaneOutboundIP = azure.PublicIPSpec{
+			Name: azure.GenerateControlPlaneOutboundIPName(s.ClusterName()),
+		}
+	} else {
+		controlPlaneOutboundIP = azure.PublicIPSpec{
+			Name:    s.APIServerPublicIP().Name,
+			DNSName: s.APIServerPublicIP().DNSName,
+			IsIPv6:  false, // currently azure requires a ipv4 lb rule to enable ipv6
+		}
+	}
+
 	return []azure.PublicIPSpec{
+		controlPlaneOutboundIP,
 		{
 			Name: azure.GenerateNodeOutboundIPName(s.ClusterName()),
 		},
-		{
-			Name:    s.Network().APIServerIP.Name,
-			DNSName: s.Network().APIServerIP.DNSName,
-			IsIPv6:  false, // currently azure requires a ipv4 lb rule to enable ipv6
-		},
 	}
 }
 
 // LBSpecs returns the load balancer specs.
 func (s *ClusterScope) LBSpecs() []azure.LBSpec {
 	specs := []azure.LBSpec{
 		{
-			// Public API Server LB
-			Name:          azure.GeneratePublicLBName(s.ClusterName()),
-			PublicIPName:  s.Network().APIServerIP.Name,
-			APIServerPort: s.APIServerPort(),
-			Role:          infrav1.APIServerRole,
+			// Control Plane LB
+			Name:              s.APIServerLB().Name,
+			SubnetName:        s.ControlPlaneSubnet().Name,
+			FrontendIPConfigs: s.APIServerLB().FrontendIPs,
+			APIServerPort:     s.APIServerPort(),
+			Type:              s.APIServerLB().Type,
+			Role:              infrav1.APIServerRole,
+			BackendPoolName:   azure.GenerateBackendAddressPoolName(s.APIServerLB().Name),
 		},
 		{
 			// Public Node outbound LB
-			Name:         s.ClusterName(),
-			PublicIPName: azure.GenerateNodeOutboundIPName(s.ClusterName()),
-			Role:         infrav1.NodeOutboundRole,
+			Name: s.NodeOutboundLBName(),
+			FrontendIPConfigs: []infrav1.FrontendIP{
+				{
+					Name: azure.GenerateFrontendIPConfigName(s.NodeOutboundLBName()),
+					PublicIP: &infrav1.PublicIPSpec{
+						Name: azure.GenerateNodeOutboundIPName(s.ClusterName()),
+					},
+				},
+			},
+			Type:            infrav1.Public,
+			BackendPoolName: azure.GenerateOutboundBackendAddressPoolName(s.NodeOutboundLBName()),
```

Comment:
n.b.: https://github.com/kubernetes-sigs/cluster-api-provider-azure/compare/master...alexeldeib:ace/integrate?expand=1#diff-fcb822faaca21eede09f649536848e46d7789e73a2ba9f2ad1b50ebc1d298ed7R167

https://github.com/kubernetes-sigs/cluster-api-provider-azure/compare/master...alexeldeib:ace/integrate?expand=1#diff-bfd4ab4960c565313b05443fc66a48e60a79f9866f6a39fad03044b2161ac69eR192-R194

It'll make my life a little smoother in the future if you don't use generate + LB name on the scope, here but attach the pool name directly to the scope :)

Reply:
I moved it to the describer but didn't take it out of scope since everything else is there right now, we can lift and shift everything as part of #825